modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/CreateDumpFutureTask.java

```diff
@@ -444,12 +452,14 @@ public PartitionDumpContext(CacheGroupContext gctx, int part, ConcurrentMap<Long
                 for (int cache : gctx.cacheIds())
                     changed.put(cache, new GridConcurrentHashSet<>());
 
-                File dumpFile = new File(groupDirectory(gctx), PART_FILE_PREFIX + part + DUMP_FILE_EXT);
+                String fileName = PART_FILE_PREFIX + part + DUMP_FILE_EXT;
+
+                File dumpFile = new File(groupDirectory(gctx), zip ? fileName + ZIP_SUFFIX : fileName);
 
                 if (!dumpFile.createNewFile())
                     throw new IgniteException("Dump file can't be created: " + dumpFile);
 
-                file = ioFactory.create(dumpFile);
+                file = zip ? new ZipFileIO(dumpFile) : ioFactory.create(dumpFile);
```

Comment:
It seems we must use another `ioFactory` when `compress==true`.
We can add `.zip` extension in it, also.